src/tokenizedSubaccounts/PPTSA.sol

```diff
@@ -0,0 +1,559 @@
+// SPDX-License-Identifier: GPL-3.0-only
+pragma solidity ^0.8.20;
+
+import {IntLib} from "lyra-utils/math/IntLib.sol";
```

Comment:
A lot of copy-paste here from the CC vault. Some slight differences in some additional validations. Check the details of this PR for more info on some of the validations

Reply:
I tried to leave comments where I thought I had some unknowns. I may have missed a few spots so please call out any red flags as I am new to solidity and this codebase.